mutt/charset.c

```diff
@@ -303,25 +304,22 @@ int mutt_ch_convert_nonmime_string(char **ps)
   if (!ps)
     return -1;
 
-  const char *c1 = NULL;
+  const struct ListNode *np = NULL;
 
-  for (const char *c = C_AssumedCharset; c; c = c1 ? c1 + 1 : 0)
+  STAILQ_FOREACH(np, &C_AssumedCharset->head, entries)
   {
+    const char *c = (const char*) np->data;
     char *u = *ps;
     size_t ulen = mutt_str_strlen(*ps);
 
     if (!u || !*u)
       return 0;
 
-    c1 = strchr(c, ':');
-    size_t n = c1 ? c1 - c : mutt_str_strlen(c);
+    size_t n = mutt_str_strlen(c);
     if (n == 0)
       return 0;
-    char *fromcode = mutt_mem_malloc(n + 1);
-    mutt_str_strfcpy(fromcode, c, n + 1);
     char *s = mutt_str_substr_dup(u, u + ulen);
-    int m = mutt_ch_convert_string(&s, fromcode, C_Charset, 0);
-    FREE(&fromcode);
+    int m = mutt_ch_convert_string(&s, c, C_Charset, 0);
     FREE(&s);
     if (m == 0)
```

Comment:
While you're altering this function, please can you change:
- `m` to `rc` (return code) for consistency
- `c` to `from` (maybe)
- `s` to `str` (maybe)

so that when the reader gets to `int m = mutt_ch_convert_string(&s, c, C_Charset, 0);`
they have an idea what's happening.